File: src/main/java/org/droidupnp/model/upnp/CallableRendererFilter.java
package org.droidupnp.model.upnp;

public class CallableRendererFilter implements ICallableFilter {

	private IUPnPDevice device;

	public void setDevice(IUPnPDevice device)
	{
		this.device = device;
	}

	@Override
	public Boolean call() throws Exception
	{
		return device.asService("RenderingControl");
	}
}

File: upnp/src/main/java/com/m3sv/plainupnp/upnp/CallableRendererFilter.kt
package com.m3sv.plainupnp.upnp

import com.m3sv.plainupnp.data.upnp.UpnpDevice
import com.m3sv.plainupnp.upnp.CallableFilter

class CallableRendererFilter : CallableFilter {

    override var device: UpnpDevice? = null

    override fun call(): Boolean = device?.asService("RenderingControl") ?: false
}
